README.md

```diff
@@ -1,27 +1,74 @@
-* If 'Maven Integration' plugin is disabled add path variable 'MAVEN_REPOSITORY' pointing to '<USER_HOME>/.m2/repository' directory (File | Settings | Appearance & Behavior | Path Variables).
-* Use Build | Build Project to build the code.
-
-To run the IDE from the built code
-* use the provided shared run configuration "**IDEA**".
-
-To run tests apply these setting to the **default** JUnit run configuration type:
-* Working dir: 
-  `<IDEA_HOME>\bin`
-* VM parameters: 
-  `-ea -Xmx128M -Djava.system.class.loader=com.intellij.util.lang.UrlClassLoader -Didea.config.path=../test-config -Didea.system.path=../test-system`
-  
-To build an installation packages, run `ant` command in `<IDEA_HOME>` directory. See build.xml file for details.  
-
-You can find other useful information at [http://www.jetbrains.org](http://www.jetbrains.org). [Contribute section](http://www.jetbrains.org/display/IJOS/Contribute) of that site describes how you can contribute to IntelliJ IDEA.
\ No newline at end of file
+These instructions will help you build IntelliJ IDEA Community Edition from source code, which is the basis for IntelliJ Platform development. The following conventions will be used to refer to
+directories on your machine:
+* `<USER_HOME>` is your home directory.
+* `<IDEA_HOME>` is the root directory for the IntelliJ source code.
+* `<JDK_16_HOME>` is the root directory for the 1.6 JDK, which is optional.
+* `<JDK_18_HOME>` is the root directory for the 1.8 JDK.
+
+## Getting IntelliJ IDEA Community Edition Source Code
+IntelliJ IDEA Community Edition source code is available from
+`github.com/JetBrains/intellij-community` by either cloning or
+downloading a zip file (based on a branch) into `<IDEA_HOME>`. The
+default is the *master* branch. 
+
+The master branch contains the source code for the most recent public release of IntelliJ IDEA. The branch names and build numbers for older
+releases of IntelliJ IDEA can be found on the page of
+[Build Number Ranges](http://www.jetbrains.org/intellij/sdk/docs/basics/getting_started/build_number_ranges.html).
+
+If you intend to make open source contributions to the IntelliJ
+Platform, see [Contributing to the IntelliJ Project](http://www.jetbrains.org/display/IJOS/Contribute) for more information.
+
+_**Speed Tip:**_ If the complete repository history isn't
+needed then using a shallow clone (`git clone --depth 1`) will save significant time.
+
+These Git operations can also be done through the [IntelliJ IDEA user interface](https://www.jetbrains.com/help/idea/using-git-integration.html).
+
+IntelliJ IDEA Community Edition requires additional Android modules from
+separate Git repositories. To clone these repositories, run one of the
+`getPlugins` scripts located in the `<IDEA_HOME>` directory. Note: by default these 
+scripts clone their respective *master* branches. You must ensure the branch/tags used for cloning
```

Comment:
> used for cloning

This looks like we suggest to edit these scripts to clone a different branch/tag, but this may be not very convenient. I think the simplest way is cloning 'master' in all the three repositories and then switching the branches in all repositories at once via 'Branches' popup in the IDE.

Reply:
My bad, text confuses clone and checkout.